### PR TITLE
mcs: fix scheduler memory sync in api server

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/7388

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/pdctl/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -165,14 +200,12 @@ func (suite *schedulerTestSuite) checkScheduler(cluster *tests.TestCluster) {
 			"transfer-witness-leader-scheduler": true,
 			"balance-witness-scheduler":         true,
 		}
-		checkSchedulerCommand(args, expected)
 
 		// check update success
-		// FIXME: remove this check after scheduler config is updated
```
Comment (lhy1024) [inline tests/pdctl/scheduler/scheduler_test.go:171]: It is removed.

File: pkg/schedule/schedulers/base_scheduler.go [hunk 2]
```diff
@@ -92,8 +92,8 @@ func (s *BaseScheduler) GetNextInterval(interval time.Duration) time.Duration {
 	return intervalGrow(interval, MaxScheduleInterval, exponentialGrowth)
 }
 
-// Prepare does some prepare work
-func (s *BaseScheduler) Prepare(cluster sche.SchedulerCluster) error { return nil }
+// ConfigPrepare does some prepare work about config.
+func (s *BaseScheduler) ConfigPrepare(cluster sche.SchedulerCluster) error { return nil }
```
Comment (rleungx) [inline pkg/schedule/schedulers/base_scheduler.go:96]: How about prepareConfig?

Comment (lhy1024) [reply pkg/schedule/schedulers/base_scheduler.go:96]: ok

**Review by rleungx — APPROVED** [review]
The rest LGTM.

File: pkg/schedule/schedulers/base_scheduler.go [hunk 3]
```diff
@@ -92,8 +92,8 @@ func (s *BaseScheduler) GetNextInterval(interval time.Duration) time.Duration {
 	return intervalGrow(interval, MaxScheduleInterval, exponentialGrowth)
 }
 
-// ConfigPrepare does some prepare work about config.
-func (s *BaseScheduler) ConfigPrepare(cluster sche.SchedulerCluster) error { return nil }
+// PrepareConfig does some prepare work about config.
+func (s *BaseScheduler) PrepareConfig(cluster sche.SchedulerCluster) error { return nil }
 
 // ConfigCleanup does some cleanup work about config.
 func (s *BaseScheduler) ConfigCleanup(cluster sche.SchedulerCluster) {}
```
Comment (rleungx) [inline pkg/schedule/schedulers/base_scheduler.go:99]: ditto

**Pull request merged**